src/client/Microsoft.Identity.Client/Extensibility/AcquireTokenForClientBuilderExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Identity.Client.Extensibility
+{
+
+    /// <summary>
+    /// Extensions for <see cref="AcquireTokenForClientBuilderExtensions"/> class
+    /// </summary>
+    public static class AcquireTokenForClientBuilderExtensions
+    {
+        /// <summary>
+        /// Binds the token to a key in the cache. L2 cache keys contain the key id.
+        /// No cryptographic operations is performed on the token.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="keyId">A key id to which the access token is associated. The token will not be retrieved from the cache unless the same key id is presented. Can be null.</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static AcquireTokenForClientParameterBuilder WithKeyId(
```

Comment:
Can this method be merged into that new `WithClientAssertion` as a param? Right now both methods would have to be called. This is similar situation that we had with the long-running OBO work, separate WithCacheKey method, kind of didn't work out.

Reply:
These are not really related. One is about specifying a custom client assertion, and one is about taking over the POP kid.
Also, POP is only relevant for AcquireTokenForClient now.